geode-core/src/test/java/org/apache/geode/metrics/internal/InternalDistributedSystemMetricsServiceBuilderTest.java

```diff
@@ -0,0 +1,287 @@
+    CollectingServiceLoader<MetricsPublishingService> theServiceLoader =
+        mock(CollectingServiceLoader.class);
+
+    serviceBuilder.setMetricsServiceFactory(metricsServiceFactory)
+        .setServiceLoader(theServiceLoader)
+        .build(system);
+
+    verify(metricsServiceFactory)
+        .create(any(), any(), same(theServiceLoader), any(), any(), any(), any(), anyBoolean());
+  }
+
+  // Warning suppressed here because no way to suppress it on the call to any(Class)
+  @SuppressWarnings("unchecked")
+  @Test
+  public void constructsServiceLoader_ifServiceLoaderNotSet() {
+    serviceBuilder
+        .setMetricsServiceFactory(metricsServiceFactory)
+        .build(system);
+
+    verify(metricsServiceFactory)
+        .create(any(), any(), any(CollectingServiceLoader.class), any(), any(), any(), any(),
+            anyBoolean());
+  }
+
+  @Test
+  public void passesGivenCompositeMeterRegistryToFactoryAsMetricsServiceMeterRegistry() {
+    CompositeMeterRegistry theMetricsServiceMeterRegistry = new CompositeMeterRegistry();
+
+    serviceBuilder.setMetricsServiceFactory(metricsServiceFactory)
+        .setCompositeMeterRegistry(theMetricsServiceMeterRegistry)
+        .build(system);
+
+    verify(metricsServiceFactory)
+        .create(any(), any(), any(), same(theMetricsServiceMeterRegistry), any(), any(), any(),
+            anyBoolean());
+  }
+
+  @Test
+  public void constructsCompositeMeterRegistry_ifMetricsServiceMeterRegistryNotSet() {
+    serviceBuilder.setMetricsServiceFactory(metricsServiceFactory)
+        .build(system);
+
+    verify(metricsServiceFactory)
+        .create(any(), any(), any(), any(CompositeMeterRegistry.class), any(), any(), any(),
+            anyBoolean());
+  }
+
+  @Test
+  public void passesIndividuallyAddedClientMeterRegistriesToFactory() {
+    Set<MeterRegistry> individuallyAddedClientMeterRegistries = setOf(3, MeterRegistry.class);
+
+    serviceBuilder.setMetricsServiceFactory(metricsServiceFactory);
+    individuallyAddedClientMeterRegistries.forEach(serviceBuilder::addPersistentMeterRegistry);
+    serviceBuilder.build(system);
+
+    @SuppressWarnings("unchecked")
+    ArgumentCaptor<Collection<MeterRegistry>> clientMeterRegistriesPassedToFactory =
+        ArgumentCaptor.forClass(Collection.class);
+
+    verify(metricsServiceFactory)
+        .create(any(), any(), any(), any(), clientMeterRegistriesPassedToFactory.capture(), any(),
+            any(), anyBoolean());
+
+    assertThat(clientMeterRegistriesPassedToFactory.getValue())
+        .containsExactlyInAnyOrderElementsOf(individuallyAddedClientMeterRegistries);
+  }
+
+  @Test
+  public void passesBulkAddedClientMeterRegistriesToFactory() {
+    Set<MeterRegistry> bulkAddedClientMeterRegistries = setOf(4, MeterRegistry.class);
+
+    serviceBuilder.setMetricsServiceFactory(metricsServiceFactory)
+        .addPersistentMeterRegistries(bulkAddedClientMeterRegistries)
+        .build(system);
+
+    @SuppressWarnings("unchecked")
```

Comment:
I know you and Aaron use these SuppressWarnings annotations quite a bit, but in general I would prefer to avoid it. I would just let warnings be warnings and we strive to minimize them but not suppress them.

Reply:
I'll fix this. (After all, if you turn on warnings about `@SuppressWarnings` annotations, this just trades one warning for another.)